homeassistant/components/decora_wifi/config_flow.py

```diff
@@ -0,0 +1,113 @@
+"""Will write later."""
+
+import logging
+
+from decora_wifi import DecoraWiFiSession
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import DOMAIN as HOMEASSISTANT_DOMAIN, HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.issue_registry import IssueSeverity, async_create_issue
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+BASE_SCHEMA = vol.Schema(
+    {
+        vol.Required("username"): str,
+        vol.Required("password"): str,
+    }
+)
+
+
+class DecoreWifiConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow for Decora Wifi Integration."""
+
+    VERSION = 1
+
+    async def async_step_import(self, import_data: dict[str, str]) -> FlowResult:
+        """Handle importting decora wifi config from configuration.yaml."""
+
+        async_create_issue(
+            self.hass,
+            HOMEASSISTANT_DOMAIN,
+            f"deprecated_yaml_{DOMAIN}",
+            breaks_in_ha_version="2024.6.0",
+            is_fixable=False,
+            issue_domain=DOMAIN,
+            severity=IssueSeverity.WARNING,
+            translation_key="deprecated_yaml",
+            translation_placeholders={
+                "domain": DOMAIN,
+                "integration_title": "Decora Wifi",
+            },
+        )
+
+        if CONF_USERNAME not in import_data or CONF_PASSWORD not in import_data:
```

Comment:
Also, the schema still works, and in the schema, username and password are required, so no need to worry about this

Reply:
removed
https://github.com/home-assistant/core/pull/104573/commits/bdc54d73df3c1a04ec5d118cf6a0bbeb204001de